ClientCacheThroughHazelcastInstanceTest.getCache_whenOwnerInstanceIsShutdown_thenOperateOnCacheFails

Failed on the PR builder of #18650 (Although the PR is against the 4.2.z branch, the same problem applies to the 5.0)

[Jenkins URL
](http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/4937/testReport/junit/com.hazelcast.client.cache.instance/ClientCacheThroughHazelcastInstanceTest/getCache_whenOwnerInstanceIsShutdown_thenOperateOnCacheFails/)

com.hazelcast.client.cache.instance.ClientCacheThroughHazelcastInstanceTest.getCache_whenOwnerInstanceIsShutdown_thenOperateOnCacheFails


Stacktrace:
```
java.lang.AssertionError: Expected exception type: class com.hazelcast.client.HazelcastClientNotActiveException, but actual exception type: class com.hazelcast.spi.exception.TargetDisconnectedException
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.cache.instance.CacheThroughHazelcastInstanceTest.whenOwnerInstanceIsShutdown_thenOperateOnCacheFails(CacheThroughHazelcastInstanceTest.java:352)
	at com.hazelcast.cache.instance.CacheThroughHazelcastInstanceTest.getCache_whenOwnerInstanceIsShutdown_thenOperateOnCacheFails(CacheThroughHazelcastInstanceTest.java:324)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)```

--- a/hazelcast/src/test/java/com/hazelcast/client/cache/instance/ClientCacheThroughHazelcastInstanceTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/cache/instance/ClientCacheThroughHazelcastInstanceTest.java
@@ -31,7 +31,10 @@ import org.junit.runner.RunWith;
 
 import javax.cache.spi.CachingProvider;
 
+import java.util.UUID;
+
 import static com.hazelcast.cache.CacheTestSupport.createClientCachingProvider;
+import static com.hazelcast.client.test.ClientTestSupport.makeSureDisconnectedFromServer;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
@@ -83,7 +86,9 @@ public class ClientCacheThroughHazelcastInstanceTest extends CacheThroughHazelca
     @Override
     protected void shutdownOwnerInstance(HazelcastInstance instance) {
         if (ownerInstance != null) {
+            UUID memberUuid = ownerInstance.getLocalEndpoint().getUuid();
             ownerInstance.shutdown();
+            makeSureDisconnectedFromServer(instance, memberUuid);
         } else {
             throw new IllegalStateException("");
         }

--- a/hazelcast/src/test/java/com/hazelcast/client/test/ClientTestSupport.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/test/ClientTestSupport.java
@@ -79,12 +79,12 @@ public class ClientTestSupport extends HazelcastTestSupport {
         ((TestClientRegistry.MockTcpClientConnectionManager) connectionManager).unblockTo(address);
     }
 
-    protected HazelcastClientInstanceImpl getHazelcastClientInstanceImpl(HazelcastInstance client) {
+    protected static HazelcastClientInstanceImpl getHazelcastClientInstanceImpl(HazelcastInstance client) {
         HazelcastClientProxy clientProxy = (HazelcastClientProxy) client;
         return clientProxy.client;
     }
 
-    protected void makeSureDisconnectedFromServer(final HazelcastInstance client, UUID memberUUID) {
+    public static void makeSureDisconnectedFromServer(final HazelcastInstance client, UUID memberUUID) {
         assertTrueEventually(() -> {
             ClientConnectionManager connectionManager = getHazelcastClientInstanceImpl(client).getConnectionManager();
             assertNull(connectionManager.getConnection(memberUUID));

